### PR TITLE
Jivarofad/m2560 update

### PR DESCRIPTION
**Pull request by JivaroFAD**
Mega2560 board and voice drivers

### CODE REVIEW

**Comment by JivaroFAD** [comment]
@bsongis : please could you give me your advice about the merge conflict on lcd.cp ?
I replace : `#if defined(BOLD_FONT) && !defined(CPUM64) || defined(TELEMETRY_NONE)`
by : `#if defined(BOLD_SPECIFIC_FONT)` (because defined in radio/src/fonts.h)
to be more maintenable.
Do you agree ?

File: radio/src/fonts.h [hunk 1]
```diff
@@ -36,7 +36,7 @@ void loadFontCache();
 extern const pm_uchar font_5x7[];
 extern const pm_uchar font_10x14[];
 
-#if defined(BOLD_FONT) && (!defined(CPUM64) || defined(TELEMETRY_NONE)) && !defined(BOOT)
```
Comment (bsongis) [inline radio/src/fonts.h]: Why this change? Isn't the option BOLD=OFF enough?

File: radio/src/gui/128x64/lcd.cpp [hunk 2]
```diff
@@ -519,7 +519,7 @@ void lcdDrawNumber(coord_t x, coord_t y, lcdint_t val, LcdFlags flags, uint8_t l
         x += 2;
       }
     }
-#if defined(BOLD_FONT) && !defined(CPUM64) || defined(TELEMETRY_NONE)
```
Comment (bsongis) [inline radio/src/gui/128x64/lcd.cpp]: Why?

File: radio/src/gui/128x64/lcd.cpp [hunk 3]
```diff
@@ -591,7 +591,7 @@ void lcdDrawNumber(coord_t x, coord_t y, lcdint_t val, LcdFlags flags, uint8_t l
     if (dblsize && (lcduint_t)val >= 1000 && (lcduint_t)val < 10000) x-=2;
     val = qr.quot;
     x -= fw;
-#if defined(BOLD_FONT) && !defined(CPUM64) || defined(TELEMETRY_NONE)
```
Comment (bsongis) [inline radio/src/gui/128x64/lcd.cpp]: Why?

File: radio/src/fonts.h (hunk 1 above)
Comment (JivaroFAD) [inline radio/src/fonts.h]: I'm not the author of this line.
I propose only to replace it by something more understandable.

Comment (bsongis) [inline radio/src/fonts.h]: Why don't you keep it as it is now?

File: radio/src/gui/128x64/lcd.cpp (hunk 2 above)
Comment (bsongis) [inline radio/src/gui/128x64/lcd.cpp]: Here the same

File: radio/src/gui/128x64/lcd.cpp (hunk 3 above)
Comment (bsongis) [inline radio/src/gui/128x64/lcd.cpp]: The same her

File: radio/src/fonts.h (hunk 1 above)
Comment (JivaroFAD) [inline radio/src/fonts.h]: I made it because, actualy, if we need to modify the condition for defining BOLD_SPECIFIC_FONT, we don't have to forget to do the same in lcd.cpp.
With this mod, there is only one modification to do, at the "root" of the font definition, and the consequences follow naturaly.

Comment (JivaroFAD) [inline radio/src/fonts.h]: Correction done right now

**Comment by bsongis** [comment]
Seems all good, hay!

**Comment by Ingwie** [comment]
Hi,

I see here my code (audio drivers for examples) from https://github.com/Ingwie/NextStepRc-2.18/tree/master/radio/src/targets/mega2560

-> without any reference to the real author (git address only please).

@jivarofad,  copy my code with a reference to my git please (https://www.gnu.org/licenses/old-licenses/gpl-2.0.en.html)

**Comment by JivaroFAD** [comment]
You alredeay have merge **our** little collaboration work to OpenTX one year ago here (https://github.com/opentx/opentx/commit/5f65217124c8715a33fc18611f275c5413ce1ec2), without any personal revendication at that time.
Second point : the git you mention, created during our collaboration, was based on my work and deposited without my autorization and, however, with the name of my brand "NextStepRC". At the end of this collaboration, we have decided together that you transfer me this git to continue my project by myself, you never did it. So, I have merged this work to OpenTX, as I inform you and the community  [here](https://www.modelisme.com/forum/aero-radio-open-source/197308-nextsteprc-radio-opentx-diy-mmag-juin-2015-a-197.html#post2536903) and as you already did one year ago

**Comment by bsongis** [comment]
Thanks @JivaroFAD for your answer, I missed some context for this one :)
